src/snowflake/snowpark/functions.py

```diff
@@ -10987,6 +10987,84 @@ def make_interval(
     return res
 
 
+@publicapi
+def make_ym_interval(
+    years: Optional[ColumnOrName] = None,
+    months: Optional[ColumnOrName] = None,
+    _emit_ast: bool = True,
+) -> Column:
+    """
+    Creates a year-month interval expression using with specified years and months.
+
+    This YearMonthInterval is not to be confused with the interval created by make_interval.
+    You can define a table column to be of data type YearMonthIntervalType.
+
+    Args:
+        years: The number of years, positive or negative
+        months: The number of months, positive or negative
+
+    Returns:
+        A Column representing a year-month interval
+
+    Example::
+
+        >>> from snowflake.snowpark.functions import make_ym_interval
+        >>>
+        >>> df = session.create_dataframe([[1, 2]], ["years", "months"])
+        >>> df.select(make_ym_interval(col("years"), col("months")).alias("interval")).show()
+        --------------
+        |"INTERVAL"  |
+        --------------
+        |+1-02       |
+        --------------
+        <BLANKLINE>
+
+    """
+    years_col = lit(0) if years is None else _to_col_if_str(years, "make_ym_interval")
+    months_col = (
+        lit(0) if months is None else _to_col_if_str(months, "make_ym_interval")
+    )
+
+    total_months = years_col * lit(12) + months_col
+
+    normalized_years = iff(
+        total_months >= lit(0),
+        cast(floor(total_months / lit(12)), "int"),
+        cast(floor((total_months * lit(-1)) / lit(12)), "int"),
+    )
+
+    normalized_months = iff(
+        total_months >= lit(0),
+        cast(total_months % lit(12), "int"),
+        cast((lit(12) * normalized_years + total_months) * lit(-1), "int"),
+    )
```

Comment:
Ideally shouldn't this logic be moved to server-side? Is it possible to move this logic there?